torch/_inductor/autotune_process.py

```diff
@@ -261,22 +363,27 @@ def worker() -> float:
         return out
 
 
+class TestBenchmarkRequest(BenchmarkRequest):
```

Comment:
sorry, why do we need to do this ? would be nice to keep separation between test and source code if possible.

One idiom that might be helpful is runtime patching objects in test code, e.g. `torch._inductor.autotune_process.BenchmarkRunner = TestBenchmarkRequest`

Reply:
@eellison, sure lemme try your idiom and see if that solves my problem. My guess is that it won't, tho. The problem that I was running into was that the sub-process didn't know how to unpickle a `TestBenchmarkRequest` if that class is defined in the test file (which is unknown to this file). I think we'd still have that problem. Another option is to define the Test class in the test file (where it belongs), but simply import it here, along with a lint directive to ignore and not remove that import, perhaps? That struck me as equally hacky. Yet another option is dispense with this special test class altogether and manipulate real BenchmarkRequest objects in the unit test. That's how I started, but the logic was so much more convoluted than the test implementation you see in this diff -- I liked the simplicity of the test. But I can definitely try harder to get a tidy-looking test without putting this here if you prefer that route.

Reply:
@eellison, yeah docs describe it better than I do: https://docs.python.org/3/library/pickle.html#pickle-picklable

> Note that functions (built-in and user-defined) are pickled by fully [qualified name](https://docs.python.org/3/glossary.html#term-qualified-name), not by value. [2](https://docs.python.org/3/library/pickle.html#id8) This means that only the function name is pickled, along with the name of the containing module and classes. Neither the function’s code, nor any of its function attributes are pickled. Thus the defining module must be importable in the unpickling environment, and the module must contain the named object, otherwise an exception will be raised.

So the crux of the test is replacing the benchmark() function with something simple than can exercise the obj.benchmark() call without actually doing the real benchmarking. The module unpickling needs to know that function definition.